soroban-env-host/src/native_contract/token/allowance.rs

```diff
@@ -63,23 +66,28 @@ pub fn write_allowance(
     // Returns the allowance to write and the previous expiration of the existing allowance.
     // If an allowance didn't exist, then the previous expiration will be None.
     let allowance_with_old_expiration_option: Option<(AllowanceValue, Option<u32>)> =
-        if let Ok(allowance) = e.get_contract_data(key.try_into_val(e)?, StorageType::Temporary) {
-            let mut updated_allowance: AllowanceValue = allowance.try_into_val(e)?;
-            updated_allowance.amount = amount;
+        match StorageUtils::try_get(e, key.try_into_val(e)?, StorageType::Temporary)? {
+            Some(allowance) => {
+                let mut updated_allowance: AllowanceValue = allowance.try_into_val(e)?;
+                updated_allowance.amount = amount;
 
-            let old_expiration = updated_allowance.expiration_ledger;
-            updated_allowance.expiration_ledger = expiration;
-            Some((updated_allowance, Some(old_expiration)))
-        } else if amount > 0 {
-            Some((
-                AllowanceValue {
-                    amount,
-                    expiration_ledger: expiration,
-                },
-                None,
-            ))
-        } else {
-            None
+                let old_expiration = updated_allowance.expiration_ledger;
+                updated_allowance.expiration_ledger = expiration;
+                Some((updated_allowance, Some(old_expiration)))
+            }
+            None => {
```

Comment:
nit: this could be somewhat simplified to `if let Some(allowance) = ... {} else if amount > 0 {} else {}` (basically what you had before, but with `Some` instead of `Ok`). This applies to all the other places as well, but here I think it's especially beneficial as it reduces the nesting level.

Reply:
I wasn't sure if it was a good idea to encourage the use of `if let Some` in case someone in the future thinks `if let Ok` works here as well (since the bug is pretty subtle), but it's definitely the "rusty" thing to do, so I made the change.

Reply:
I think the current interface is clear enough - the main caveat is really to use the helper instead of just `get`.